autofish.lua

```diff
@@ -0,0 +1,220 @@
+
+Arguments
+-------------
+-r, --include-raw
+  Allow the stock tracking to also count raw fish
+enable
+  Enable this script
+disable
+  Disable this script
+status
+  Print the current status
+max
+  The maximum number of fish you want to stop fishing at.
+min
+  the minimum number of fish before restarting fishing.
+]====]
+-- set default enabled state
+enabled = enabled or false
+set_maxFish = set_maxFish or 100
+set_minFish = set_minFish or 50
+set_useRaw = set_useRaw or false
+isFishing = isFishing or true
+
+function isEnabled()
+    return enabled
+end
+
+local function persist_state()
+    persist.GlobalTable[GLOBAL_KEY] = json.encode({enabled=enabled,
+        set_maxFish=set_maxFish, set_minFish=set_minFish, set_useRaw=set_useRaw,
+        isFishing=isFishing
+    })
+end
+
+local function load_state()
+    -- load persistent data
+    local persisted_data = json.decode(persist.GlobalTable[GLOBAL_KEY] or "")
+    enabled = (persisted_data or {enabled=false})["enabled"]
+    set_maxFish = (persisted_data or {set_maxFish=100})["set_maxFish"]
+    set_minFish = (persisted_data or {set_minFish=50})["set_minFish"]
+    set_useRaw = (persisted_data or {set_useRaw=false})["set_useRaw"]
+    isFishing = (persisted_data or {isFishing=true})["isFishing"]
+end
+
+
+-- toggle the fishing labour on all dwarves/work detail if enabled
+function toggle_fishing_labour(state)
+    -- pass true to state to turn on, otherwise disable
+    -- find all work details that have fishing enabled:
+    local work_details = df.global.plotinfo.hauling.work_details
+    for k,v in pairs(work_details) do
+        if v.allowed_labors.FISH then
+            -- set limited to true just in case a custom work detail is being
+            -- changed, to prevent *all* dwarves from fishing.
+            v.work_detail_flags.limited = true
+            v.work_detail_flags.enabled = state
+        end
+    end
+    isFishing = state -- save current state
+end
+
+-- check if an item isn't forbidden/rotten/on fire/etc..
+function isValidItem(item)
+    local flags = item.flags
+    if flags.rotten or flags.trader or flags.hostile or flags.forbid
+        or flags.dump or flags.on_fire or flags.garbage_collect then
+        return false
+    end
+    return true
+end
+
+function event_loop()
+    if not enabled then return end
+    local world = df.global.world
+
+    -- count the number of valid fish we have. (not rotten, forbidden, on fire, dumping...)
+    local prepared, raw = 0, 0
+    for k,v in pairs(world.items.other[df.items_other_id.IN_PLAY]) do
+        if v:getType() == df.item_type.FISH and isValidItem(v) then
+            prepared = prepared + v:getStackSize()
+        end
+        if (v:getType() == df.item_type.RAW_FISH and isValidItem(v)) and set_useRaw then
+            raw = raw + v:getStackSize()
+        end
+    end
+
+    -- hande pausing/resuming labour
+    local sumFish = prepared + raw
```

Comment:
if this is written as `numFish = set_useRaw and (prepared + raw) or prepared` then you can consolidate the logic below.